profiling/src/profiling.rs

```diff
@@ -727,102 +727,42 @@ impl Profiler {
                     locals.last_cpu_time = Some(now);
                 }
 
-                let mut labels = vec![];
-                let gpc = datadog_php_profiling_get_profiling_context;
-                if let Some(get_profiling_context) = gpc {
-                    let context = get_profiling_context();
-                    if context.local_root_span_id != 0 {
-                        labels.push(Label {
-                            key: "local root span id".into(),
-                            value: LabelValue::Str(
-                                format!("{}", context.local_root_span_id).into(),
-                            ),
-                        });
-                        labels.push(Label {
-                            key: "span id".into(),
-                            value: LabelValue::Str(format!("{}", context.span_id).into()),
-                        });
-                    }
-                }
-
-                let mut tags = locals.tags.clone();
-
-                if let Some(version) = PHP_VERSION.get() {
-                    /* This should probably be "language_version", but this is
-                     * the tag that was standardized for this purpose. */
-                    let tag = Tag::new("runtime_version", version)
-                        .expect("runtime_version to be a valid tag");
-                    tags.push(tag);
-                }
-
-                if let Some(sapi) = crate::SAPI.get() {
-                    match Tag::new("php.sapi", sapi.to_string()) {
-                        Ok(tag) => tags.push(tag),
-                        Err(err) => warn!("Tag error: {}", err),
-                    }
-                }
-
-                let message = SampleMessage {
-                    key: ProfileIndex {
-                        sample_types,
-                        tags,
-                        endpoint: locals.uri.clone(),
-                    },
-                    value: SampleData {
-                        frames,
-                        labels: labels.clone(),
-                        sample_values,
-                    },
-                };
+                let message = self.prepare_sample_message(
+                    frames,
+                    interrupt_count,
+                    walltime,
+                    cputime,
+                    0,
+                    0,
+                    locals,
+                );
 
                 match self.send_sample(message) {
-                    Ok(_) => trace!(
-                        "Sent stack sample of depth {} with labels {:?} to profiler.",
-                        depth,
-                        labels
-                    ),
+                    Ok(_) => trace!("Sent stack sample of depth {} with to profiler.", depth,),
                     Err(err) => warn!(
-                        "Failed to send stack sample of depth {} with labels {:?} to profiler: {}",
-                        depth, labels, err
```

Comment:
NIT: These changes look unrelated to allocation profiling on first glance. Would probably be easier to split this PR into smaller ones if it contains multiple orthogonal changes.